### ISSUE TITLE
Duplicating result on constituency_parser

### ISSUE THREAD

**Issue by skulick2** [allenai/allennlp#2612]
Hi, 

The models page states for the Constituency Parser that "This model uses ELMo embeddings, which are completely character based and improves single model performance from 92.6 F1 to 94.11 F1 on the Penn Tree bank, a 20% relative error reduction."  I would to duplicate these results on the Penn Treebank before training on a different dataset.  However, I have not been able to do so, and I'm wondering what training parameters were used to get that 92.6 number (which is also a bit higher than in the Stern et al paper the parser is based on)

  I used constituency_parser.jsonnet in the training_config directory of the github, together with the standard sections 2-21(train),22 (dev), 23(test) of the Penntreebank.  Using the Penntreebank with gold part of speech tags, and modifying the jsonnet to evaluate_on_test, I got:

"test_evalb_f1_measure": 0.9049651918761865
which is not just lower than the 92.6, but also lower than in the Stern et al paper (91.8). 

(I've attached the complete log as constituency_parser-stdout.log)

I also tried training on the version of the penn treebank with non-gold part of speech tags, as done for the Stern et al paper, and I got an even lower score 
"test_evalb_f1_measure": 0.8848756906077349

However, training and testing with constituency_parser_elmo.jsonnet gave a score of 
test_evalb_f1_measure": 0.9434116724085002
which is actually a bit higher than what is said on the models page.

 (in all cases the score on the validation(dev) section was extremely close to that on the test section.)

Is there a config file available that will result in the score reaching that of the Stern et al paper, or the higher number reported on the models page?  I have not yet gone through the Stern implementation to figure out what might  be different.  

Many thanks in advance for help with this!

[constituency_parser-stdout.log](https://github.com/allenai/allennlp/files/2973021/constituency_parser-stdout.log)

**Comment by DeNeutoy** [comment]
Hi there,

I'm not sure why that config doesn't use glove embeddings - it should do, try the 100 dim ones. The only other difference I recorded with this config is a batch size of 10 - this gave me 91.2 Dev F1. Also, replacing the LSTM with `stacked_bidirectional_lstm` should get you the rest of the way. Note that this is not an exact replication of the hyperparameters of the stern et al parser, as their original one took considerably longer to train.

**Comment by DeNeutoy** [comment]
Re-open if you have more questions.
